### PR TITLE
feat:  Add policy for detecting GitHub Secret-Scanning

### PR DESCRIPTION
**Pull request by naphtalid-legit**
* Add collection of 'Security and Analysis' settings for github repos

#### Check off the following:

- [x] This PR follows the CONTRIBUTION.md guidelines
- [x] I have self-reviewed my changes before submitting the PR

### CODE REVIEW

File: policies/github/repository.rego [hunk 1]
```diff
@@ -716,4 +716,23 @@ repository_secret_is_stale[stale] := true{
         "update date" : time.format(secret.updated_at),
     }
 
+}
+
+# METADATA
+# scope: rule
+# title: Secret Scanning should be enabled for public repositories
+# description: Public repositories should not have secrets hard-coded in the files. It is recommended to enable some sort of secret scanning.
```
Comment (noamd-legit) [inline policies/github/repository.rego:724]: Need to rephrase the policy desc, the issue applies for private repositories as well

File: internal/collectors/github/repository_collector.go [hunk 2]
```diff
@@ -407,6 +412,23 @@ func (rc *repositoryCollector) withSecrets(repository ghcollected.Repository, lo
...
 }
 
+func (rc *repositoryCollector) withSecurityAndAnalysis(repo ghcollected.Repository, login string) (ghcollected.Repository, error) {
+	
+	securityAndAnalysis, err := rc.Client.GetSecurityAndAnalysisForRepository(repo.Name(), login)
+	if err != nil {
+		return repo, err
+	}
+	if securityAndAnalysis == nil {
+		perm := collectors.NewMissingPermission(permissions.RepoAdmin, collectors.FullRepoName(login, repo.Repository.Name),
```
Comment (noamd-legit) [inline internal/collectors/github/repository_collector.go:422]: Did you verify it for "no permissions" and "doesn't have branch protection"?

File: internal/analyzers/skippers/skippers.go [hunk 3]
```diff
@@ -39,6 +39,14 @@ func NewSkipper(ctx context.Context) Skipper {
 			"enterprise": func(_ collectors.CollectedData) bool {
 				return !context_utils.GetIsCloud(ctx)
 			},
+			"is_branch_protection_supported": func(data collectors.CollectedData) bool {
```
Comment (noamd-legit) [inline internal/analyzers/skippers/skippers.go:42]: Please combine this logic and "has_branch_protection_permissions" to a single skipper. and update the old policies to use it.

**Review by noamd-legit — COMMENTED** [review]
Add unit and e2e tests

File: internal/analyzers/skippers/skippers.go (hunk 3 above)
Comment (naphtalid-legit) [reply internal/analyzers/skippers/skippers.go:42]: But there may be policies that we don't have permissions but BP is supported, so we need the other skipper for them.

File: policies/github/repository.rego [hunk 4]
```diff
@@ -716,4 +716,23 @@ repository_secret_is_stale[stale] := true{
         "update date" : time.format(secret.updated_at),
     }
 
+}
+
+# METADATA
+# scope: rule
+# title: Secret Scanning should be enabled
+# description: Repositories should not have secrets hard-coded in the files. It is recommended to enable some sort of secret scanning.
```
Comment (noamd-legit) [inline policies/github/repository.rego:724]: Should be phrased as a recommendation "Repository should have secret scanning enabled, reason...."

File: internal/collectors/github/repository_collector.go [hunk 5]
```diff
@@ -444,16 +461,38 @@ func (rc *repositoryCollector) fixBranchProtectionInfo(repository ghcollected.Re
...
-func (rc *repositoryCollector) checkMissingPermissions(repo ghcollected.Repository, entityName string) []collectors.MissingPermission {
+func (rc *repositoryCollector) checkMissingPermissions(repo ghcollected.Repository, entityName string, repoContext *repositoryContext) []collectors.MissingPermission {
 	var missingPermissions []collectors.MissingPermission
 	if repo.NoBranchProtectionPermission {
 		effect := "Cannot read repository branch protection information"
 		perm := collectors.NewMissingPermission(permissions.RepoAdmin, entityName, effect, namespace.Repository)
 		missingPermissions = append(missingPermissions, perm)
 	}
+	if repo.SecurityAndAnalysis == nil {
+		//if (not org.owner and not scope.repo_admin) {
```
Comment (noamd-legit) [inline internal/collectors/github/repository_collector.go:472]: remove comment

File: internal/clients/github/client.go [hunk 6]
```diff
@@ -656,3 +657,16 @@ func (c *Client) GetOrganizationSecrets(org string) (*gh.Secrets, error) {
 	}
 	return secrets, nil
 }
+
+func (c *Client) GetSecurityAndAnalysisForRepository(repo, owner string) (*gh.SecurityAndAnalysis, error) {
```
Comment (noamd-legit) [inline internal/clients/github/client.go:661]: check if we can add this to the graphql query

**Review by noamd-legit — APPROVED** [review]

File: internal/clients/github/client.go (hunk 6 above)
Comment (naphtalid-legit) [reply internal/clients/github/client.go:661]: Didn't find something in GraphQL that gives info about this

**Pull request merged**